src/sentry/eventstream/kafka/consumer.py

```diff
@@ -0,0 +1,183 @@
+                i.partition,
+                i.offset if i.offset > -1 else self._get_initial_offset(consumer, i),
+            ) for i in assignment
+        ]
+
+        self._consumer.assign(assignment)
+
+        for i in assignment:
+            # Setting the local offsets will either cause the partition to be
+            # paused (if the remote offset is unknown or the local offset is
+            # not trailing the remote offset) or resumed.
+            self._partition_state_manager.set_local_offset(i.topic, i.partition, i.offset)
+
+    def _rewind_partitions_on_assignment(self, consumer, assignment):
+        # The commit log consumer must start consuming from the beginning of
+        # the commit log topic to ensure that it has a comprehensive view of
+        # all active partitions.
+        consumer.assign([TopicPartition(i.topic, i.partition, OFFSET_BEGINNING)
+                         for i in assignment])
+
+    def get_commit_data(self, message):
+        group, topic, partition = message.key().decode('utf-8').split(':', 3)
+        partition = int(partition)
+        offset = int(message.value().decode('utf-8'))
+        return group, topic, partition, offset
+
+    def start(self):
+        self._consumer = Consumer({
+            'bootstrap.servers': self.bootstrap_servers,
+            'group.id': self.consumer_group,
+            'enable.auto.commit': 'false',
+            'enable.auto.offset.store': 'true',
+            'enable.partition.eof': 'false',
+            'default.topic.config': {
+                'auto.offset.reset': 'error',
+            },
+            'on_commit': self._commit_callback,
+        })
+
+        self._partition_state_manager = SynchronizedPartitionStateManager(
+            self._on_partition_state_change)
+
+        self._consumer.subscribe(self.topics, self._pause_partitions_on_assignment)
+
+        # NOTE: The commit log consumer group should not be persisted into the
+        # ``__consumer_offsets`` topic since no offsets are committed by this
+        # consumer. The group membership metadata messages will be published
+        # initially but as long as this group remains a single consumer it will
+        # be deleted after the consumer is closed. It is very important to note
+        # that the ``group.id`` **MUST** be unique to this consumer process!!!
+        # (This ensures that it is able to consume from all partitions of the
+        # commit log topic and get a comprehensive view of the state of the
+        # consumer groups it is tracking.)
+        self._commit_log_consumer = Consumer({
+            'bootstrap_servers': self.bootstrap_servers,
+            'group.id': '{}:sync:{}'.format(self.consumer_group, uuid.uuid1.hex()),
+            'enable.auto.commit': 'false',
+            'enable.auto.offset.store': 'true',
+            'enable.partition.eof': 'false',
+            'default.topic.config': {
+                'auto.offset.reset': 'error',
```

Comment:
Is `auto.offset.reset` even relevant here as this consumer does not commit any offsets, and so can't really be reset to anything?

Reply:
I'm not 100% sure how offset resets are handled for a live consumer (if it managed to either consume slowly enough that it was trying to read messages that were dropped due to retention, or somehow got ahead of the stream, or some other sort of bug that caused it to read invalid positions) but other than that, I don't think there's really any reason it'd need to be set to `error`.